### PR TITLE
Add gnu-complexity 1.5

### PR DESCRIPTION
**Pull request by iankronquist**
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

_You can erase any parts of this template not applicable to your Pull Request._

### New Formulae Submissions:

- [x] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

Reopening #44336. I fixed all the relevant comments there including adding a test.

### CODE REVIEW

**Comment by apjanke** [comment]
Test is failing. Looks like it's failing to load a library from `autogen`. Maybe `autogen` needs to be a full-time dependency instead of just `:build`?

```
failed: brew test --verbose gnu-complexity

Stacktrace

        Testing gnu-complexity
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20160302-46309-1n11cgs.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/test.rb /usr/local/Library/Formula/gnu-complexity.rb --verbose
==> /usr/local/Cellar/gnu-complexity/1.5/bin/complexity -t 3 ./test.c
dyld: Library not loaded: /usr/local/opt/autogen/lib/libopts.25.dylib
  Referenced from: /usr/local/Cellar/gnu-complexity/1.5/bin/complexity
  Reason: image not found
Error: gnu-complexity: failed
Failed executing: /usr/local/Cellar/gnu-complexity/1.5/bin/complexity -t 3 ./test.c
/usr/local/Library/Homebrew/formula.rb:1482:in `block in system'
/usr/local/Library/Homebrew/formula.rb:1419:in `open'
/usr/local/Library/Homebrew/formula.rb:1419:in `system'
/usr/local/Library/Formula/gnu-complexity.rb:44:in `block in <class:GnuComplexity>'
/usr/local/Library/Homebrew/formula.rb:1328:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:37:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:1324:in `run_test'
/usr/local/Library/Homebrew/test.rb:28:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/test.rb:27:in `<main>'
```

**Comment by iankronquist** [comment]
Interesting, that's not a dependency I would have expected. I removed it from the build group, let's see if it works.

File: Library/Formula/gnu-complexity.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  desc "Measures complexity of C source"
+  homepage "https://www.gnu.org/software/complexity"
+  url "http://ftpmirror.gnu.org/complexity/complexity-1.5.tar.xz"
+  mirror "https://ftp.gnu.org/gnu/complexity/complexity-1.5.tar.xz"
+  sha256 "1f5194c0dc6e813ea1dc7f36bfd05f15786a8ad6e9b3ab65b53d2263a0d93102"
+
+  depends_on "autogen"
+
+  def install
+    system "./configure", "--disable-debug",
```
Comment (DomT4) [inline Library/Formula/gnu-complexity.rb]: `--disable-debug` wasn't recognised last time. Is it now? Ref: https://github.com/Homebrew/homebrew/pull/44336#discussion_r41718770

File: Library/Formula/gnu-complexity.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+class GnuComplexity < Formula
+  desc "Measures complexity of C source"
+  homepage "https://www.gnu.org/software/complexity"
+  url "http://ftpmirror.gnu.org/complexity/complexity-1.5.tar.xz"
+  mirror "https://ftp.gnu.org/gnu/complexity/complexity-1.5.tar.xz"
+  sha256 "1f5194c0dc6e813ea1dc7f36bfd05f15786a8ad6e9b3ab65b53d2263a0d93102"
+
+  depends_on "autogen"
```
Comment (DomT4) [inline Library/Formula/gnu-complexity.rb]: If needed at runtime should be `depends_on "autogen" => :run`

Comment (iankronquist) [inline Library/Formula/gnu-complexity.rb]: Sorry about that. Should be fixed. Let's see if the tests pass.

File: Library/Formula/gnu-complexity.rb (hunk 1 above)
Comment (iankronquist) [inline Library/Formula/gnu-complexity.rb]: It seems to be? When I built it locally I didn't see the configure script complain. If you see something different, let me know.

This is also a newer version.

Comment (iankronquist) [inline Library/Formula/gnu-complexity.rb]: Oops, forgot to pass the `-v` flag. No, it is not recognized, so I removed it.

**Comment by iankronquist** [comment]
@DomT4 Addressed your comments.

File: Library/Formula/gnu-complexity.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/gnu-complexity.rb]: No worries. It's a style nit on our part really; these are almost always `:build`-only dependencies so we label them explicitly when they aren't so it doesn't look like they were left runtime accidentally.

File: Library/Formula/gnu-complexity.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  def install
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.").write <<-EOS.undent
```
Comment (DomT4) [inline Library/Formula/gnu-complexity.rb]: Should this be `(testpath/"test.c")`? That's what your calling below.

**Comment by iankronquist** [comment]
Is there a way to run the tests for this formula so I can catch errors like that?
Sorry about the back and forth here, this is the first formula I've submitted.

**Comment by dunn** [comment]
> Is there a way to run the tests for this formula so I can catch errors like that?

Yeah, locally you can do `brew test --verbose --sandbox <formula>`.

**Comment by iankronquist** [comment]
Okay, thanks. It looks like the ci is green. Is there anything else you'd like?

**Comment by dunn** [comment]
All good, thanks! Merged in a148dcfe1c5691b125c1aa5169aba83ad5b37ef6.

**Pull request closed without merging**